app/src/main/java/com/github/orelzion/spotifyclone/view/TracksListFragment.kt

```diff
@@ -0,0 +1,82 @@
+package com.github.orelzion.spotifyclone.view
+
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import com.github.orelzion.spotifyclone.R
+import com.github.orelzion.spotifyclone.model.Tracks
+import com.github.orelzion.spotifyclone.model.TracksResponse
+import com.github.orelzion.spotifyclone.model.repository.BrowseRepository
+import com.github.orelzion.spotifyclone.viewmodel.TrackViewData
+
+
+/**
+ * A fragment representing a list of Tracks.
+ */
+class TracksListFragment : Fragment(R.layout.fragment_items_list) {
+    private val tracksListView: RecyclerView by lazy { requireView().findViewById(R.id.itemsList) }
+    private val tracksAdapter: TracksAdapter by lazy { TracksAdapter() }
+
+    /**
+     *
+     */
+    companion object {
+        /**
+         * factory method
+         * accessed via TracksListFragment, as it is being declared in companion object
+         */
+        fun newInstance(albumId: String): TracksListFragment {
+            return TracksListFragment().apply {
+                arguments = Bundle().apply {
+                    putString("albumId", albumId)
+                    // TODO: Ask Orel - Shall I declare albumId as a Pair? in/outside companion object?
```

Comment:
I'm not sure I get why we need a pair here

Reply:
Pair for key and value